### DISCUSSION TITLE
SideBar Navigation with NextJS

### DISCUSSION THREAD

**Discussion by ebitogu** [vercel/next.js#26920] [Help]
I love NextJS file based routing but I have a situation where I need to update a component section and update the browser route without refreshing.

What I want to achieve is what React-Router has done here in [This Link](https://reactrouter.com/web/example/sidebar).

Please, how can I achieve this with NextJs?

Thank you.

**Comment by ICE-Cold-Ethanol** [comment]
A nice article about this topic:
[Persistent Layout Patterns in Next.js](https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/)
